backend-lib/src/main/rust/lib.rs

```diff
@@ -1326,58 +1317,44 @@ pub unsafe extern "C" fn Java_cash_z_ecc_android_sdk_internal_jni_RustBackend_de
     unwrap_exc_or(&env, res, JNI_FALSE)
 }
 
-fn zip317_helper<Ctx, DbT, R>(
-    context: Ctx,
+fn zip317_helper<DbT>(
+    change_memo: Option<MemoBytes>,
     use_zip317_fees: jboolean,
-    enabled: impl FnOnce(Ctx, GreedyInputSelector<DbT, zip317::SingleOutputChangeStrategy>) -> R,
-    disabled: impl FnOnce(Ctx, GreedyInputSelector<DbT, fixed::SingleOutputChangeStrategy>) -> R,
-) -> R {
-    if use_zip317_fees == JNI_TRUE {
-        let input_selector = GreedyInputSelector::new(
-            zip317::SingleOutputChangeStrategy::new(zip317::FeeRule::standard()),
-            DustOutputPolicy::default(),
-        );
-
-        enabled(context, input_selector)
+) -> GreedyInputSelector<DbT, SingleOutputChangeStrategy> {
+    let fee_rule = if use_zip317_fees == JNI_TRUE {
+        StandardFeeRule::Zip317
     } else {
-        let input_selector = GreedyInputSelector::new(
-            fixed::SingleOutputChangeStrategy::new(fixed::FeeRule::standard()),
-            DustOutputPolicy::default(),
-        );
-
-        disabled(context, input_selector)
-    }
+        #[allow(deprecated)]
+        StandardFeeRule::PreZip313
+    };
+    GreedyInputSelector::new(
+        SingleOutputChangeStrategy::new(fee_rule, change_memo),
+        DustOutputPolicy::default(),
+    )
 }
 
 #[no_mangle]
-pub unsafe extern "C" fn Java_cash_z_ecc_android_sdk_internal_jni_RustBackend_createToAddress(
+pub unsafe extern "C" fn Java_cash_z_ecc_android_sdk_internal_jni_RustBackend_proposeTransfer(
     env: JNIEnv<'_>,
     _: JClass<'_>,
     db_data: JString<'_>,
-    usk: jbyteArray,
+    account: jint,
     to: JString<'_>,
     value: jlong,
     memo: jbyteArray,
-    spend_params: JString<'_>,
-    output_params: JString<'_>,
     network_id: jint,
     use_zip317_fees: jboolean,
 ) -> jbyteArray {
```

Comment:
nit: It might be nice to have a type alias for `jbyteArray` that indicates that this is a serialized proposal protobuf. The fact that the return type didn't change was a little disorienting.